life_expectancy/cleaning.py

```diff
@@ -21,93 +20,134 @@
 
 # Define variables cleaning  and filtering data
 COMPOSED_COL = 'unit,sex,age,geo\\time'
-DECOMPOSED_COLs = ['unit', 'sex', 'age','region']
+DECOMPOSED_COLS= ['unit', 'sex', 'age','region']
 
-def clean_data(region_filter: str) -> None:
+def load_data (input_file_path: Path = INPUT_FILE_PATH) -> pd.DataFrame:
     """
-    This function processes and filters a file with european life expectancy data over the years.
+    This function loads a file with european life expectancy data over the years.
 
     Parameters:
-    None
+    input_file_path (str): The path to the input file.
 
     Returns:
-    Just saves the processed data for Portugal.
+    df_raw (pd.DataFrame): The loaded data.
     """
+    df_raw = pd.DataFrame()  # define an empty DataFrame
 
     try:
         # load data
-        with open(INPUT_FILE_PATH, encoding='utf-8') as filename:
+        with open(input_file_path, encoding='utf-8') as filename:
             df_raw = pd.read_csv(filename, sep='\t', na_values=[':'])
-    except FileNotFoundError:# pragma: no cover
-        logging.error("Input file %s not found.", INPUT_FILE_PATH)
-        return
+    except PermissionError:# pragma: no cover
+        logging.error("Permission denied to read %s.", input_file_path)
+        return df_raw
+    except FileNotFoundError as error:# pragma: no cover
+        logging.error(str(error))
+        return df_raw
+    except pd.errors.ParserError as error:# pragma: no cover
+        logging.error("Error reading file %s: %s", input_file_path, error)
+        return df_raw
+    return df_raw
```

Comment:
```suggestion
return pd.read_csv(filename, sep='\t', na_values=[':'], encoding="utf-8")
```

Eu diria que há certo overengineering aqui. 

A inicialização com um DataFrame vazio é desnecessária. O método `read_csv` já cria um DataFrame. Ao "inicializar" um DataFrame vazio, você está na verdade criando um segundo dataframe na tua memória. Esse dataframe pendurará como "garbage value" após a linha 40.

A unica vantagem de termos o DataFrame vazio é retornamos algo quando a erro de leitura do csv... mas será mesmo isso que queremos? Não seria melhor pararmos o programa em vez de continuarmos com um DataFrame vazio?

Reply:
alterei como sugerido assumindo que é feita uma exceção de erro pelo pandas qdo na conseguimos ler o csv, faz sentido?